File: indexrecyclerview/src/main/java/com/banzhi/indexrecyclerview/interfaces/IDataHelper.java
package com.banzhi.indexrecyclerview.interfaces;

import com.banzhi.indexrecyclerview.bean.BaseIndexBean;

import java.util.List;

/**
 * <pre>
 * @author : No.1
 * @time : 2018/8/2.
 * @desciption :
 * @version :
 * </pre>
 */

public interface IDataHelper {
    /**
     * 数据转换 根据getorderName生成pinyin
     *
     * @param datas
     */
    void cover(List<? extends BaseIndexBean> datas);

    /**
     * 排序
     *
     * @param datas
     */
    void sortDatas(List<? extends BaseIndexBean> datas);

    /**
     * 排序并获取索引数据
     *
     * @param datas
     */
    void sortDatasAndGetIndex(List<? extends BaseIndexBean> datas, List<String> indexDatas);

    /**
     * 获取索引
     *
     * @param datas
     * @param indexDatas
     */
    void getIndex(List<? extends BaseIndexBean> datas, List<String> indexDatas);
}

File: indexrecyclerview/src/main/java/com/banzhi/indexrecyclerview/interfaces/IDataHelper.kt
package com.banzhi.indexrecyclerview.interfaces

import com.banzhi.indexrecyclerview.bean.BaseIndexBean

/**
 * <pre>
 * @author :
 * @time : 2021/5/26.
 * @desciption :
 * @version :
</pre> *
 */
interface IDataHelper {
    /**
     * 数据转换 根据getorderName生成pinyin
     *
     * @param datas
     */
    fun cover(datas: MutableList<out BaseIndexBean>)

    /**
     * 排序
     *
     * @param datas
     */
    fun sortDatas(datas: MutableList<out BaseIndexBean>)

    /**
     * 排序并获取索引数据
     *
     * @param datas
     */
    fun sortDatasAndGetIndex(datas: MutableList<out BaseIndexBean>, indexDatas: MutableList<String>)

    /**
     * 获取索引
     *
     * @param datas
     * @param indexDatas
     */
    fun getIndex(datas: MutableList<out BaseIndexBean>, indexDatas: MutableList<String>)
}